upb/msg_internal.h

```diff
@@ -82,14 +125,25 @@ typedef struct {
   _upb_field_parser *field_parser;
 } _upb_fasttable_entry;
 
+typedef union {
+  const struct upb_msglayout *submsg;
+  // TODO: const upb_enumlayout *subenum;
+} upb_msglayout_sub;
```

Comment:
another place where a longer name might help.  upb_submessage_layout is a bit easier to read for me.  That said, this one might want to wait for us to try and rename upb_message_layout or something